addons/mail/models/mail_thread.py

```diff
@@ -1644,12 +1644,23 @@ def _message_partner_info_from_emails(self, emails, link_mail=False):
 
     def _message_post_process_attachments(self, attachments, attachment_ids, message_values):
         """ Preprocess attachments for mail_thread.message_post() or mail_mail.create().
+        Purpose is to
+
+          * transfer attachments given by ``attachment_ids`` from the composer to
+            the record (if any);
+          * limit attachments manipulation when being a shared user;
+          * create attachments from ``attachments``. If those are linked to the
+            content (body) through CIDs body is updated accordingly;
+
+        :param list(tuple(str,str), tuple(str,str, dict)) attachments : list of attachment
+            tuples in the form ``(name,content)`` or ``(name,content, info)`` where content
+            is NOT base64 encoded;
+        :param list(id, id) attachment_ids: list of existing attachments to link to this message
+        :param msg_vals: dictionary of values used to create the message. If given
```

Comment:
Wrong parameter name and also strange explanation.
It looks copy/pasted from other places, the name of the parameters is "message_values" here and it is not really "optional to avoid accessing self", it's very much required (and has apparently nothing to do with self?).

Reply:
Arf indeed, wrong copy paste actually.